base/shared/src/main/scala/scalaz/data/IStream.scala

```diff
@@ -0,0 +1,22 @@
+package scalaz
+package data
+
+import scalaz.control.Inf
+
+
+trait IStreamModule {
+  type IStream[A]
+
+  def scons[A](a: =>A)(as: =>Id[IStream[A]]): IStream[A]
+  def uncons[A](as: =>IStream[A]): (Inf[A], Inf[IStream[A]])
```

Comment:
`uncons` should return an actual `A` (as opposed to a thunk of `A`).

Again, if `Stream` already encapsulates laziness, there should be no need to wrap it in `Inf`.